library/multichoicerecyclerview/src/main/java/com/davidecirillo/multichoicerecyclerview/MultiChoiceAdapter.java

```diff
@@ -80,8 +83,14 @@ public void selectAll() {
      * @return true if has been selected false otherwise
      */
     public boolean select(int position) {
-        if (mIsInMultiChoiceMode || mIsInSingleClickMode && mItemList.containsKey(position)) {
-            perform(Action.SELECT, position, true, true);
+        if (mItemList.containsKey(position)) {
+            if (mIsInSingleClickMode || mIsInMultiChoiceMode) {
+                perform(Action.SELECT, position, true, true);
+                return true;
+            }
+        } else {
+            mItemList.put(position, State.INACTIVE);
```

Comment:
Yeah, changing that. I've opened an issue for that #49 I'll take a look at this later.